mono/utils/mono-threads-api.h

```diff
@@ -22,34 +22,54 @@ MONO_BEGIN_DECLS
 This API is experimental. It will eventually be required to properly use the rest of the raw-omp embedding API.
 */
 
+// FIXME This is duplicated in libtest.c and mono-threads-api.h.
+typedef struct _MonoStackData {
+	const char *function_name;
+} MonoStackData;
+typedef const MonoStackData *MonoStackDataParam; // usual
+typedef       MonoStackData *MonoStackDataParamMutable; // rare
+// const here seems to trigger compiler bug, giving static instead of stack storage
+#define MONO_STACK_DATA(x) /*const*/ MonoStackData  x##_value = { __func__ }; \
+                           const MonoStackDataParam x = &x##_value
+
 MONO_API gpointer
-mono_threads_enter_gc_unsafe_region (gpointer* stackdata);
+mono_threads_enter_gc_unsafe_region (MonoStackDataParam stackdata);
 
 MONO_API void
-mono_threads_exit_gc_unsafe_region (gpointer cookie, gpointer* stackdata);
+mono_threads_exit_gc_unsafe_region (gpointer cookie, MonoStackDataParam stackdata);
 
 MONO_API gpointer
-mono_threads_enter_gc_unsafe_region_unbalanced (gpointer* stackdata);
+mono_threads_enter_gc_unsafe_region_unbalanced (MonoStackDataParam stackdata);
 
-MONO_API void
-mono_threads_exit_gc_unsafe_region_unbalanced (gpointer cookie, gpointer* stackdata);
+gpointer
+ves_icall_mono_threads_enter_gc_unsafe_region_unbalanced (MonoStackDataParamMutable stackdata);
```

Comment:
The `ves_icall_` should not be in this file. Add them to mono-threads.h

Reply:
The name will change, but probably the move is desired the same, ok.

Reply:
I put them in this file so that mono_foo and similar ves_icall_mono_foo are adjacent but ok.